src/plugins/intel_gpu/tests/unit/test_cases/stft_gpu_test.cpp

```diff
@@ -0,0 +1,208 @@
+        ret.frameSize = helpers::AllocateTensor<int64_t>({}, {testParam.frameSize});
+
+        return ret;
+    }
+
+    void Execute(const STFTInferenceParams& params) {
+        // Prepare the network.
+        auto stream = get_test_stream_ptr(get_test_default_config(engine_));
+
+        auto scalar_layout = params.frameSize->get_layout();
+        scalar_layout.set_partial_shape({});
+
+        topology topology;
+        topology.add(input_layout("signal", params.signal->get_layout()));
+        topology.add(input_layout("window", params.window->get_layout()));
+        topology.add(input_layout("frameSize", scalar_layout));
+        topology.add(input_layout("frameStep", scalar_layout));
+        topology.add(STFT("stft",
+                          input_info("signal"),
+                          input_info("window"),
+                          input_info("frameSize"),
+                          input_info("frameStep"),
+                          params.transposedFrames));
+
+        cldnn::network::ptr network = get_network(engine_, topology, get_test_default_config(engine_), stream, false);
+
+        network->set_input_data("signal", params.signal);
+        network->set_input_data("window", params.window);
+        network->set_input_data("frameSize", params.frameSize);
+        network->set_input_data("frameStep", params.frameStep);
+
+        // Run and check results.
+        auto outputs = network->execute();
+
+        auto output = outputs.at("stft").get_memory();
+
+        helpers::CompareBuffers(output, params.expectedOutput, get_test_stream());
+    }
+
+private:
+    engine& engine_ = get_test_engine();
+};
+
+std::vector<STFTTestParams> generateTestParams() {
+    std::vector<STFTTestParams> params;
+#define TEST_DATA(signalShape,                        \
+                  windowShape,                        \
+                  outputShape,                        \
+                  frameSize,                          \
+                  frameStep,                          \
+                  transposedFrames,                   \
+                  signalData,                         \
+                  windowData,                         \
+                  expectedOutput,                     \
+                  testcaseName)                       \
+    params.push_back(STFTTestParams{signalShape,      \
+                                    windowShape,      \
+                                    outputShape,      \
+                                    frameSize,        \
+                                    frameStep,        \
+                                    transposedFrames, \
+                                    signalData,       \
+                                    windowData,       \
+                                    expectedOutput,   \
+                                    testcaseName});
+
+#include "unit_test_utils/tests_data/stft_data.h"
```

Comment:
Can we reuse ov::reference::stft() implementation from src/core/reference/include/openvino/reference/stft.hpp here?

Reply:
Not sure what you meant by that. If you want to compare output of gpu kernel directly to  ov::reference::stft(), than this is exactly what happens(to my best knowledge) in "functional tests", which are implemented in this PR(see openvino/src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/stft.cpp).

Reply:
Yes, both tests aim to compare the kernel's results with certain reference values. However, I don't think the origin of these reference data is important here. So, if there is no difference between them, I'd prefer to use ov::reference::stft() call over some hardcoded values, simply because it offers more flexibility and supportability :)

Reply:
Honestly I disagree. That data was generated from pytorch and in this form it can be reused in other tests. Ref impl can be buggy more likely pytroch(I already had such issue). Please note that I have applied the same pattern in e.g. SearchSorted op

Reply:
Ok, I see your point, let's keep this
However these type of issues should probably be caught at another validation level, rather than in GPU kernels test with hardcoded values